src/Service.Tests/Unittests/MultiSourceQueryExecutionUnitTests.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Azure.DataApiBuilder.Config;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Azure.DataApiBuilder.Core.Configurations;
+using Azure.DataApiBuilder.Core.Models;
+using Azure.DataApiBuilder.Core.Resolvers;
+using Azure.DataApiBuilder.Core.Resolvers.Factories;
+using Azure.DataApiBuilder.Core.Services;
+using Azure.DataApiBuilder.Service.GraphQLBuilder.Directives;
+using Azure.DataApiBuilder.Service.GraphQLBuilder.GraphQLTypes;
+using Azure.DataApiBuilder.Service.Tests.GraphQLBuilder.Helpers;
+using HotChocolate;
+using HotChocolate.Execution;
+using HotChocolate.Execution.Processing;
+using HotChocolate.Resolvers;
+using HotChocolate.Types;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace Azure.DataApiBuilder.Service.Tests.Unittests
+{
+    [TestClass]
+    public class MultiSourceQueryExecutionUnitTests
+    {
+        string _query = @"{
+                clients_by_pk(Index: 3) {
+                    FirstName
+                }
+                customers_by_pk(Index: 3) {
+                    Name
+                }
+            }";
+
+        /// <summary>
+        /// Validates successful execution of a query against multiple sources.
```

Comment:
you don't have to enumerate each and every validation check, but can you elaborate on how this test validates successful execution of query against multiple dbs?

Reply:
Added description to testmethod.